Upstream-Status: Backport

* Backported from the ToT: https://crrev.com/c/1411333

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 8e852c23a9a2770fc6601f85f59f805102d1681b Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Wed, 16 Jan 2019 07:14:29 +0000
Subject: [PATCH 07/16] Fix ime linux export include and fix dtor.

This fixes the following error:

ERROR at //ui/base/ime/linux/fake_input_method_context_factory.h:10:11: Include not allowed.
          ^-------------------------------
It is not in any dependency of
  //ui/base/ime/linux:linux
The include file is in the target(s):
  //ui/base/ime:ime
which should somehow be reachable.

Also it overrides virtual dtor and uses "default" for
ctor and dtor.

Bug: 791920
Change-Id: I93f0036c3df94123f83f3e77854fa6373344aee4
Reviewed-on: https://chromium-review.googlesource.com/c/1411333
Reviewed-by: Shu Chen <shuchen@chromium.org>
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#623144}
---
 ui/base/ime/linux/fake_input_method_context_factory.cc | 4 +++-
 ui/base/ime/linux/fake_input_method_context_factory.h  | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ui/base/ime/linux/fake_input_method_context_factory.cc b/ui/base/ime/linux/fake_input_method_context_factory.cc
index f8d3102ee486..aba2f76ed304 100644
--- a/ui/base/ime/linux/fake_input_method_context_factory.cc
+++ b/ui/base/ime/linux/fake_input_method_context_factory.cc
@@ -8,7 +8,9 @@
 
 namespace ui {
 
-FakeInputMethodContextFactory::FakeInputMethodContextFactory() {}
+FakeInputMethodContextFactory::FakeInputMethodContextFactory() = default;
+
+FakeInputMethodContextFactory::~FakeInputMethodContextFactory() = default;
 
 std::unique_ptr<LinuxInputMethodContext>
 FakeInputMethodContextFactory::CreateInputMethodContext(
diff --git a/ui/base/ime/linux/fake_input_method_context_factory.h b/ui/base/ime/linux/fake_input_method_context_factory.h
index 12bb7661b1ea..1d5f8b86a789 100644
--- a/ui/base/ime/linux/fake_input_method_context_factory.h
+++ b/ui/base/ime/linux/fake_input_method_context_factory.h
@@ -7,6 +7,7 @@
 
 #include "base/macros.h"
 #include "ui/base/ime/linux/linux_input_method_context_factory.h"
+#include "ui/base/ime/linux/ui_base_ime_linux_export.h"
 
 namespace ui {
 
@@ -16,6 +17,7 @@ class UI_BASE_IME_LINUX_EXPORT FakeInputMethodContextFactory
     : public LinuxInputMethodContextFactory {
  public:
   FakeInputMethodContextFactory();
+  ~FakeInputMethodContextFactory() override;
 
   // LinuxInputMethodContextFactory:
   std::unique_ptr<LinuxInputMethodContext> CreateInputMethodContext(
-- 
2.17.1

